bigquery_etl/cli/query.py

```diff
@@ -1262,6 +1265,151 @@ def render(name, sql_dir, output_dir):
             click.echo(rendered_sql)
 
 
+def _parse_partition_date(partition_date):
+    params = partition_date.split(":")
+    if len(params) != 3:
+        return None
+
+    # Check date format
+    try:
+        datetime.datetime.strptime(params[2], "%Y-%m-%d").date()
+    except ValueError:
+        return None
+
+    # Check column name
```

Comment:
I'd consider splitting column name and date partition to be two separate params. Otherwise, maybe we should consider renaming this function and it's parameter since we're not working with just a partition_date but more something like partition_settings.

Reply:
Renamed to `_parse_partition_setting`